### PR TITLE
[build-script] Consistent use of shell.capture in swift_build_support

### PR DESCRIPTION
**Pull request by rintaro**
#### What's in this pull request?

Replace `subprocess.check_call` with `shell.capture` in `swift_build_support` package.
This eliminates direct `subprocess` import. Currently, except for [`migration` module](https://github.com/apple/swift/blob/c9cd33bb0aee05c2d19cdf37f16f4f88f2c8cca2/utils/swift_build_support/swift_build_support/migration.py).

Introduced `optional` parameter to `shell.capture`.

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please Python lint

**Comment by rintaro** [comment]
@swift-ci Please Python lint

**Comment by rintaro** [comment]
@gribozavr @ddunbar Could we accept `optional` parameter?

File: utils/swift_build_support/swift_build_support/debug.py [hunk 1]
```diff
@@ -14,29 +14,31 @@
...
     Print the host machine's `xcodebuild` version, as well as version
     information for all available SDKs.
     """
-    print(u'{}\n'.format(_output(['xcodebuild', '-version'])), file=file)
-    print(u'--- SDK versions ---', file=file)
-    print(u'{}\n'.format(_output(['xcodebuild', '-version', '-sdk'])),
-          file=file)
+    version = shell.capture(
+        ['xcodebuild', '-version'],
+        dry_run=False, echo=False, optional=True).rstrip()
```
Comment (gribozavr) [inline utils/swift_build_support/swift_build_support/debug.py]: I'd rather we stop right here if the command fails.  If `xcodebuild` is not there, we're in trouble.

File: utils/swift_build_support/swift_build_support/toolchain.py [hunk 2]
```diff
@@ -151,13 +151,13 @@ def _release_date():
...
-        try:
-            # For details on `sysctl`, see:
-            # http://www.freebsd.org/cgi/man.cgi?sysctl(8)
-            out = subprocess.check_output(['sysctl', '-n', 'kern.osreldate'])
-            return int(out)
-        except subprocess.CalledProcessError:
+        # For details on `sysctl`, see:
+        # http://www.freebsd.org/cgi/man.cgi?sysctl(8)
+        out = shell.capture(['sysctl', '-n', 'kern.osreldate'],
+                            dry_run=False, echo=False, optional=True)
```
Comment (gribozavr) [inline utils/swift_build_support/swift_build_support/toolchain.py]: Similarly, I'd rather we stop than produce a subtly incorrect build.  The way I think about it is, either the user has specified an explicit FreeBSD version (e.g., for cross-compiling from a different OS), then this function shouldn't be even called, or we really need this information in order to proceed.

File: utils/swift_build_support/swift_build_support/which.py [hunk 3]
```diff
@@ -35,8 +34,8 @@ def which(cmd):
     We provide our own implementation because shutil.which() has not
     been backported to Python 2.7, which we support.
     """
-    try:
-        ret = subprocess.check_output(['which', cmd])
-        return str(ret.rstrip().decode())
-    except subprocess.CalledProcessError:
+    out = shell.capture(['which', cmd],
+                        dry_run=False, echo=False, optional=True)
+    if out is None:
```
Comment (gribozavr) [inline utils/swift_build_support/swift_build_support/which.py]: This use of `optional` LGTM.

**Comment by gribozavr** [comment]
LGTM, `optional` makes a lot of sense, but I weakly disagree about two places where it is used.  What do you think?

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by gribozavr** [comment]
@swift-ci Please smoke test

File: utils/swift_build_support/swift_build_support/debug.py (hunk 1 above)
Comment (rintaro) [inline utils/swift_build_support/swift_build_support/debug.py]: I Agree.
Moreover, we need non-Darwin version of `--show-sdks` implementation.
In this PR, I will just omit `, optional=True`

**Comment by gribozavr** [comment]
@swift-ci Please smoke test OS X platform

File: utils/swift_build_support/swift_build_support/toolchain.py (hunk 2 above)
Comment (rintaro) [inline utils/swift_build_support/swift_build_support/toolchain.py]: The docstring;

> ```
> If the release date cannot be ascertained, return None.
> ```

Honestly, I don't know much about FreeBSD, I have no clue about this decision. :(

Comment (gribozavr) [inline utils/swift_build_support/swift_build_support/toolchain.py]: OK, let's keep it as is for now.  I hope we can get some FreeBSD buildbots.

**Comment by gribozavr** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
Hm, the change list in CI looks funny.
https://ci.swift.org/job/swift-PR-osx-smoke-test/410/
I will rebase and squash on newest master.

**Comment by gribozavr** [comment]
The smoke tests are incremental, so we get a list of commits compared to the last smoke test that was run (which is irrelevant).

**Comment by gribozavr** [comment]
@swift-ci Please smoke test OS X platform

**Comment by rintaro** [comment]
Ah, got it :)

**Pull request merged**